### PR TITLE
Event create fan notifications

### PR DESCRIPTION
**Pull request by msbodetti**
### References Issues:
https://app.asana.com/0/1158142652422330/1155691976771297/f
### Release Screenshots / Video:
https://share.getcloudapp.com/5zuJkdpk
### Environment Variables:
 * No change

### CODE REVIEW

File: src/components/pages/admin/events/dashboard/fanNotifications/Index.js [hunk 1]
```diff
@@ -0,0 +1,860 @@
...
+						 sent_quantity
+					 }) => {
+						if (
+							(notification_type === "LastCall" && status === "Pending") ||
+							(notification_type === "LastCall" && status === "InProgress")
+						) {
+							notificationTriggered = true;
+						}
+
+						send_at ? this.setState({ scheduledAt: send_at }) : null;
```
Comment (mikethetike) [inline src/components/pages/admin/events/dashboard/fanNotifications/Index.js]: Isn't it better to do only one `setState()` to reduce rerendering each time?

File: src/components/pages/admin/events/dashboard/fanNotifications/Index.js [hunk 2]
```diff
@@ -0,0 +1,860 @@
...
+
+				if (
+					(response.data.notification_type === "LastCall" &&
+						response.data.status === "Pending") ||
+					(response.data.notification_type === "LastCall" &&
+						response.data.status === "InProgress")
+				) {
+					notificationTriggered = true;
+
+					this.setState({ notificationTriggered });
```
Comment (mikethetike) [inline src/components/pages/admin/events/dashboard/fanNotifications/Index.js]: Same here

File: src/components/pages/admin/events/dashboard/fanNotifications/Index.js [hunk 3]
```diff
@@ -0,0 +1,860 @@
...
+				</Button>
+			);
+		} else {
+			return null;
+		}
+	}
+
+	fifteenInterval() {
+		const { eventStart, eventEnd, times, timezone } = this.state;
+		const start = moment(eventStart, "YYYY-MM-DD hh:mm a").tz(timezone);
```
Comment (mikethetike) [inline src/components/pages/admin/events/dashboard/fanNotifications/Index.js]: Do we have constants we can use for these formats instead?

File: src/components/pages/admin/events/dashboard/fanNotifications/Index.js [hunk 4]
```diff
@@ -0,0 +1,860 @@
...
+			isNotificationAfter,
+			isEventEnded
+		} = this.state;
+		const { classes } = this.props;
+		if (scheduledAt || notificationTriggered) {
+			return (
+				<Typography className={classes.notificationBg}>
+					Notification scheduled for{" "}
+					<span className={classes.pinkText}>
+						{moment(scheduledAt).format("DD/MM/YYYY HH:mm A")}
```
Comment (mikethetike) [inline src/components/pages/admin/events/dashboard/fanNotifications/Index.js]: Probably need the venue's timezone here

**Comment by msbodetti** [comment]
Thanks @mikethetike all done :)

File: src/components/pages/admin/events/dashboard/fanNotifications/Index.js [hunk 5]
```diff
@@ -145,6 +146,7 @@ class Index extends Component {
 		this.loadEventDetails();
 
 		const notificationDate = this.state.scheduledAt
+			// eslint-disable-next-line no-undef
 			? moment(this.state.scheduledAt).tz(this.state.timezone)
```
Comment (Krakaw) [inline src/components/pages/admin/events/dashboard/fanNotifications/Index.js]: Let's extract these, e.g.
```javascript
const {scheduledAt, timezone} = this.state;
```

File: src/components/pages/admin/events/dashboard/fanNotifications/Index.js [hunk 6]
```diff
@@ -572,15 +572,17 @@ class Index extends Component {
...
 		} = this.state;
 		const { classes } = this.props;
+		const scheduledDate = moment(scheduledAt, "YYYY-MM-DD hh:mm a").tz(timezone);
 		if (scheduledAt || notificationTriggered) {
 			return (
 				<Typography className={classes.notificationBg}>
 					Notification scheduled for{" "}
 					<span className={classes.pinkText}>
-						{moment(scheduledAt).format("DD/MM/YYYY HH:mm A")}
+						{scheduledDate ? scheduledDate.format("DD/MM/YYYY HH:mm A") : ""}
```
Comment (Krakaw) [inline src/components/pages/admin/events/dashboard/fanNotifications/Index.js]: Use a const

File: src/components/pages/admin/events/dashboard/fanNotifications/Index.js [hunk 7]
```diff
@@ -505,16 +505,16 @@ class Index extends Component {
 
 	fifteenInterval() {
 		const { eventStart, eventEnd, times, timezone } = this.state;
-		const start = moment(eventStart, "YYYY-MM-DD hh:mm a").tz(timezone);
-		const end = moment(eventEnd, "YYYY-MM-DD hh:mm a").tz(timezone);
+		const start = moment(eventStart, `${TIME_FORMAT_YYYY_MM_DD_NO_TIMEZONE}`).tz(timezone);
+		const end = moment(eventEnd, `${TIME_FORMAT_YYYY_MM_DD_NO_TIMEZONE}`).tz(timezone);
```
Comment (Krakaw) [inline src/components/pages/admin/events/dashboard/fanNotifications/Index.js]: We can remove the string expansion and just place the variable in there.

File: src/components/pages/admin/events/dashboard/fanNotifications/Index.js [hunk 8]
```diff
@@ -505,16 +505,16 @@ class Index extends Component {
...
 
 		// round starting minutes up to nearest 15 (12 --> 15, 17 --> 30)
 		// note that 59 will round up to 60, and moment.js handles that correctly
 		start.minutes(Math.ceil(start.minutes() / 15) * 15);
 		while (start < end) {
 			times.push({
-				key: start.format("YYYY-MM-DDTHH:mm:ss.SSSSZ"),
-				value: start.format("DD/MM/YYYY HH:mm A")
+				key: start.format(`${TIME_FORMAT_MM_DD_YYYY_NO_TIMEZONE}`),
+				value: start.format(`${TIME_FORMAT_MM_DD_YYYY_NO_TIMEZONE}`)
```
Comment (Krakaw) [inline src/components/pages/admin/events/dashboard/fanNotifications/Index.js]: We can remove the string expansion and just place the variable in there.

**Comment by msbodetti** [comment]
> Just some minor tweaks required

All done @Krakaw :)

**Pull request merged**